projects/packages/forms/src/blocks/shared/hooks/use-form-style-outline-classes-and-styles.js

```diff
@@ -0,0 +1,136 @@
+import { isNumber, merge } from 'lodash';
+import { useFormStyle, FORM_STYLE } from '../../contact-form/util/form';
+
+/**
+ * Returns the value of the CSS var if it is a number, otherwise null.
+ * Match behaviour in projects/packages/forms/src/blocks/shared/hooks/use-jetpack-field-styles.js.
+ *
+ * @param {*} value - A value from the legacy form block attributes.
+ * @return {string|*} The value of the CSS var if it is a number, otherwise the value itself.
+ */
+function getIntAsPxValue( value ) {
+	if ( typeof value !== 'undefined' && isNumber( value ) ) {
+		return `${ value }px`;
+	}
+	return value;
+}
+
+export default function useFormStyleOutlineClassesAndStyles( {
+	clientId,
+	inputBlockName,
+	inputBlockAttributes,
+} ) {
+	const formStyle = useFormStyle( clientId );
+	const { userConfig, baseConfig } = useSelect( select => {
+		const {
+			__experimentalGetCurrentGlobalStylesId,
+			getEditedEntityRecord,
+			__experimentalGetCurrentThemeBaseGlobalStyles,
+		} = select( coreStore );
+		const globalStylesId = __experimentalGetCurrentGlobalStylesId();
+		if ( ! globalStylesId ) {
+			return null;
+		}
+
+		return {
+			userConfig: getEditedEntityRecord( 'root', 'globalStyles', globalStylesId ),
+			baseConfig: __experimentalGetCurrentThemeBaseGlobalStyles(),
+		};
+	}, [] );
+
+	const mergedGlobalBlockStyles = useMemo(
+		() =>
+			merge(
+				baseConfig?.styles?.blocks?.[ inputBlockName ],
+				userConfig?.styles?.blocks?.[ inputBlockName ]
+			),
+		[ baseConfig?.styles?.blocks, userConfig?.styles?.blocks, inputBlockName ]
+	);
+
+	return useMemo( () => {
+		// Only return styles for outlined and animated forms.
+		if ( formStyle !== FORM_STYLE.OUTLINED && formStyle !== FORM_STYLE.ANIMATED ) {
+			return null;
+		}
+		// Access the input block's attributes.
+		const blockBorderClassesAndStyles = getBorderClassesAndStyles( inputBlockAttributes ?? {} );
+		const globalBorderClassesAndStyles = getBorderClassesAndStyles( {
+			style: mergedGlobalBlockStyles,
+		} );
+
+		// Notched HTML only needs the background color and associated classes.
+		const attributesWithBackgroundColor = inputBlockAttributes
+			? {
+					backgroundColor: inputBlockAttributes?.backgroundColor,
+					style: {
+						color: {
+							background: inputBlockAttributes?.style?.color?.background,
+						},
+					},
+			  }
+			: {};
+		const blockColorClassesAndStyles = getColorClassesAndStyles( attributesWithBackgroundColor );
+
+		/**
+		 * Remove undefined classname values.
+		 */
+		const filteredBlockColorClassesAndStyles = [
+			blockBorderClassesAndStyles?.className,
+			blockColorClassesAndStyles?.className,
+		]
+			.filter( Boolean )
+			.join( ' ' );
+
+		let styleSpecificCssVars = {};
+
+		if ( formStyle === FORM_STYLE.OUTLINED ) {
+			styleSpecificCssVars = {
+				'--jetpack--contact-form--notch-width':
+					'max(var(--jetpack--contact-form--input-padding-left, 16px), var(--jetpack--contact-form--border-radius))',
```

Comment:
I think it's also a problem in trunk, but we should probably cap this value somehow. Users can set a really high value for border radius, and it makes the leading notch element will just keep increasing in size.